### ISSUE TITLE
Can't install sparse_dot_topn

### ISSUE THREAD

**Issue by JohnNevius** [ing-bank/sparse_dot_topn#28]
Windows 10, python 2.7.16:

The same (or very similar) error messages:

This was run against the most recent archive.

   ERROR: Command errored out with exit status 1:
     command: 'c:\python27\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'c:\\users\\john\\appdata\\local\\temp\\pip-install-ihe26p\\sparse-dot-topn\\setup.py'"'"'; __file__='"'"'c:\\users\\john\\appdata\\local\\temp\\pip-install-ihe26p\\sparse-dot-topn\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'c:\users\john\appdata\local\temp\pip-record-fygx8o\install-record.txt' --single-version-externally-managed --compile
         cwd: c:\users\john\appdata\local\temp\pip-install-ihe26p\sparse-dot-topn\
    Complete output (59 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-2.7
    creating build\lib.win-amd64-2.7\sparse_dot_topn
    copying sparse_dot_topn\awesome_cossim_topn.py -> build\lib.win-amd64-2.7\sparse_dot_topn
    copying sparse_dot_topn\__init__.py -> build\lib.win-amd64-2.7\sparse_dot_topn
    running build_ext
    skipping './sparse_dot_topn\sparse_dot_topn.cpp' Cython extension (up-to-date)
    skipping './sparse_dot_topn\sparse_dot_topn_threaded.cpp' Cython extension (up-to-date)
    building 'sparse_dot_topn.sparse_dot_topn' extension
    creating build\temp.win-amd64-2.7
    creating build\temp.win-amd64-2.7\Release
    creating build\temp.win-amd64-2.7\Release\sparse_dot_topn
    C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\python27\include -Ic:\python27\PC -Ic:\python27\lib\site-packages\numpy\core\include /Tp./sparse_dot_topn\sparse_dot_topn.cpp /Fobuild\temp.win-amd64-2.7\Release\./sparse_dot_topn\sparse_dot_topn.obj -Ox
    sparse_dot_topn.cpp
    c:\python27\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
    C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Include\xlocale(342) : warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
    C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\python27\include -Ic:\python27\PC -Ic:\python27\lib\site-packages\numpy\core\include /Tp./sparse_dot_topn/sparse_dot_topn_source.cpp /Fobuild\temp.win-amd64-2.7\Release\./sparse_dot_topn/sparse_dot_topn_source.obj -Ox
    sparse_dot_topn_source.cpp
    C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Include\vector(1177) : warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
            C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Include\vector(1153) : while compiling class template member function 'void std::vector<_Ty>::_Insert_n(std::_Vector_const_iterator<_Ty,_Alloc>,unsigned __int64,const _Ty &)'
            with
            [
                _Ty=candidate,
                _Alloc=std::allocator<candidate>
            ]
            ./sparse_dot_topn/sparse_dot_topn_source.cpp(67) : see reference to class template instantiation 'std::vector<_Ty>' being compiled
            with
            [
                _Ty=candidate
            ]
    C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:c:\python27\libs /LIBPATH:c:\python27\PCbuild\amd64 /LIBPATH:c:\python27\PC\VS9.0\amd64 /EXPORT:initsparse_dot_topn build\temp.win-amd64-2.7\Release\./sparse_dot_topn\sparse_dot_topn.obj build\temp.win-amd64-2.7\Release\./sparse_dot_topn/sparse_dot_topn_source.obj /OUT:build\lib.win-amd64-2.7\sparse_dot_topn\sparse_dot_topn.pyd /IMPLIB:build\temp.win-amd64-2.7\Release\./sparse_dot_topn\sparse_dot_topn.lib /MANIFESTFILE:build\temp.win-amd64-2.7\Release\./sparse_dot_topn\sparse_dot_topn.pyd.manifest
    sparse_dot_topn.obj : warning LNK4197: export 'initsparse_dot_topn' specified multiple times; using first specification
       Creating library build\temp.win-amd64-2.7\Release\./sparse_dot_topn\sparse_dot_topn.lib and object build\temp.win-amd64-2.7\Release\./sparse_dot_topn\sparse_dot_topn.exp
    building 'sparse_dot_topn.sparse_dot_topn_threaded' extension
    C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\python27\include -Ic:\python27\PC -Ic:\python27\lib\site-packages\numpy\core\include /Tp./sparse_dot_topn\sparse_dot_topn_threaded.cpp /Fobuild\temp.win-amd64-2.7\Release\./sparse_dot_topn\sparse_dot_topn_threaded.obj -Ox
    sparse_dot_topn_threaded.cpp
    c:\python27\lib\site-packages\numpy\core\include\numpy\npy_1_7_deprecated_api.h(14) : Warning Msg: Using deprecated NumPy API, disable it with #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
    C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Include\xlocale(342) : warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
    C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\python27\include -Ic:\python27\PC -Ic:\python27\lib\site-packages\numpy\core\include /Tp./sparse_dot_topn/sparse_dot_topn_source.cpp /Fobuild\temp.win-amd64-2.7\Release\./sparse_dot_topn/sparse_dot_topn_source.obj -Ox
    sparse_dot_topn_source.cpp
    C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Include\vector(1177) : warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc
            C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Include\vector(1153) : while compiling class template member function 'void std::vector<_Ty>::_Insert_n(std::_Vector_const_iterator<_Ty,_Alloc>,unsigned __int64,const _Ty &)'
            with
            [
                _Ty=candidate,
                _Alloc=std::allocator<candidate>
            ]
            ./sparse_dot_topn/sparse_dot_topn_source.cpp(67) : see reference to class template instantiation 'std::vector<_Ty>' being compiled
            with
            [
                _Ty=candidate
            ]
    C:\Users\John\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\python27\include -Ic:\python27\PC -Ic:\python27\lib\site-packages\numpy\core\include /Tp./sparse_dot_topn/sparse_dot_topn_parallel.cpp /Fobuild\temp.win-amd64-2.7\Release\./sparse_dot_topn/sparse_dot_topn_parallel.obj -Ox
    sparse_dot_topn_parallel.cpp
    ./sparse_dot_topn/sparse_dot_topn_parallel.cpp(24) : fatal error C1083: Cannot open include file: 'thread': No such file or directory
    error: command 'C:\\Users\\John\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\amd64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\python27\python.exe' -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'c:\\users\\john\\appdata\\local\\temp\\pip-install-ihe26p\\sparse-dot-topn\\setup.py'"'"'; __file__='"'"'c:\\users\\john\\appdata\\local\\temp\\pip-install-ihe26p\\sparse-dot-topn\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'c:\users\john\appdata\local\temp\pip-record-fygx8o\install-record.txt' --single-version-externally-managed --compile Check the logs for full command output.

**Comment by ymwdalex** [comment]
@JohnNevius some people in other issue suggested `conda install -c conda-forge sparse_dot_topn` in Windows? Could you please give a try? Thanks!

**Comment by Eidjaz** [comment]
I believe i am also having issues with installing this package. Not sure why but somehow i cannot bypass our internal proxy setting with this packages.

I can perform a pip install on all other packages except this, do you have any idea?



```
    ERROR: Command errored out with exit status 1:
     command: 'c:\programdata\anaconda3\python.exe' -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\eidjaz\\AppData\\Local\\Temp\\pip-install-mkdgqqz7\\sparse-dot-topn\\setup.py'"'"'; __file__='"'"'C:\\Users\\eidjaz\\AppData\\Local\\Temp\\pip-install-mkdgqqz7\\sparse-dot-topn\\setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\eidjaz\AppData\Local\Temp\pip-pip-egg-info-e6b0wxgc'
         cwd: C:\Users\eidjaz\AppData\Local\Temp\pip-install-mkdgqqz7\sparse-dot-topn\
    Complete output (24 lines):
    Download error on https://pypi.org/simple/scipy/: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond -- Some packages may not be found!
    Download error on https://pypi.org/simple/: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond -- Some packages may not be found!
    No local packages or working download links found for scipy>=1.2.3
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "C:\Users\eidjaz\AppData\Local\Temp\pip-install-mkdgqqz7\sparse-dot-topn\setup.py", line 75, in <module>
        ext_modules=[original_ext, threaded_ext],
      File "c:\programdata\anaconda3\lib\site-packages\setuptools\__init__.py", line 144, in setup
        _install_setup_requires(attrs)
      File "c:\programdata\anaconda3\lib\site-packages\setuptools\__init__.py", line 139, in _install_setup_requires
        dist.fetch_build_eggs(dist.setup_requires)
      File "c:\programdata\anaconda3\lib\site-packages\setuptools\dist.py", line 724, in fetch_build_eggs
        replace_conflicting=True,
      File "c:\programdata\anaconda3\lib\site-packages\pkg_resources\__init__.py", line 782, in resolve
        replace_conflicting=replace_conflicting
      File "c:\programdata\anaconda3\lib\site-packages\pkg_resources\__init__.py", line 1065, in best_match
        return self.obtain(req, installer)
      File "c:\programdata\anaconda3\lib\site-packages\pkg_resources\__init__.py", line 1077, in obtain
        return installer(requirement)
      File "c:\programdata\anaconda3\lib\site-packages\setuptools\dist.py", line 791, in fetch_build_egg
        return cmd.easy_install(req)
      File "c:\programdata\anaconda3\lib\site-packages\setuptools\command\easy_install.py", line 673, in easy_install
        raise DistutilsError(msg)
    distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('scipy>=1.2.3')
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by JohnNevius** [comment]
A different set of error messages:

:\Users\John>conda install -c conda-forge sparse_dot_topn
Traceback (most recent call last):
  File "C:\ProgramData\Miniconda2\Scripts\conda-script.py", line 12, in <module>
    sys.exit(main())
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\cli\main.py", line 150, in main
    return conda_exception_handler(_main, *args, **kwargs)
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\exceptions.py", line 1371, in conda_exception_handler
    return_value = exception_handler(func, *args, **kwargs)
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\exceptions.py", line 1082, in __call__
    return self.handle_exception(exc_val, exc_tb)
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\exceptions.py", line 1126, in handle_exception
    return self.handle_unexpected_exception(exc_val, exc_tb)
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\exceptions.py", line 1137, in handle_unexpected_exception
    self.print_unexpected_error_report(error_report)
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\exceptions.py", line 1207, in print_unexpected_error_report
    from .cli.main_info import get_env_vars_str, get_main_info_str
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\cli\main_info.py", line 19, in <module>
    from ..core.index import _supplement_index_with_system
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\core\index.py", line 10, in <module>
    from .package_cache_data import PackageCacheData
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\core\package_cache_data.py", line 15, in <module>
    from .path_actions import CacheUrlAction, ExtractPackageAction
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\core\path_actions.py", line 13, in <module>
    from .envs_manager import get_user_environments_txt_file, register_env, unregister_env
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\core\envs_manager.py", line 11, in <module>
    from .prefix_data import PrefixData
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\core\prefix_data.py", line 22, in <module>
    from ..common.pkg_formats.python import get_site_packages_anchor_files
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\common\pkg_formats\python.py", line 1165, in <module>
    DEFAULT_MARKER_CONTEXT = get_default_marker_context()
  File "C:\ProgramData\Miniconda2\lib\site-packages\conda\common\pkg_formats\python.py", line 1147, in get_default_marker_context
    'platform_python_implementation': platform.python_implementation(),
  File "C:\Python27\Lib\platform.py", line 1493, in python_implementation
    return _sys_version()[0]
  File "C:\Python27\Lib\platform.py", line 1455, in _sys_version
    repr(sys_version))
ValueError: failed to parse CPython sys.version: '2.7.18 |Anaconda, Inc.| (default, Apr 23 2020, 17:26:54) [MSC v.1500 64 bit (AMD64)]'

**Comment by ymwdalex** [comment]
@JohnNevius thanks for the feedback. Could you please check this post to see if it helps? https://stackoverflow.com/questions/34145861/valueerror-failed-to-parse-cpython-sys-version-after-using-conda-command

**Comment by ymwdalex** [comment]
Hi @JohnNevius, 

Thanks for the feedback! We just released [v0.3.3](https://pypi.org/project/sparse-dot-topn/#files). We employed Github actions and CiBuildWheel, and make the wheels available for Py3.6-3.10 on multiple architectures of Linux, MacOS and Windows.

If you still have installation problems, please try the latest version.

If you find any problem with v0.3.3, please let us know. Thanks!

**Comment by ymwdalex** [comment]
[v0.3.3](https://pypi.org/project/sparse-dot-topn/#files) should fix this issue
